test/cell_test.rb

```diff
@@ -34,4 +34,79 @@ def test_it_contains_a_ship
     assert_equal cruiser, cell.ship
     assert_equal false, cell.empty?
   end
+
+  def test_it_has_not_been_fired_upon_by_default
+   cruiser = Ship.new("Cruiser", 3)
+   cell = Cell.new("B4")
+
+   cell.place_ship(cruiser)
+
+   assert_equal false, cell.fired_upon?
+  end
+
+  def test_it_has_been_fired_upon
+   cruiser = Ship.new("Cruiser", 3)
+   cell = Cell.new("B4")
+   # require "pry"; binding.pry
+   cell.place_ship(cruiser)
+
+   assert_equal false, cell.fired_upon?
+
+   cell.fire_upon
+
+   assert_equal true, cell.fired_upon?
+  end
+
+  def test_it_renders_if_it_has_not_been_fired_upon
+    cell_1 = Cell.new("B4")
+
+    assert_equal ".", cell_1.render
+  end
+
+  def test_it_renders_if_it_has_been_fired_upon_but_does_not_contain_a_ship
+    cell_1 = Cell.new("B4")
+
+    cell_1.fire_upon
+
+    assert_equal "M", cell_1.render
+  end
+
+  def test_it_has_been_fired_upon_and_contains_a_ship
+    cruiser = Ship.new("Cruiser", 3)
+    cell_1 = Cell.new("B4")
+    cell_2 = Cell.new("C3")
+
+    cell_2.place_ship(cruiser)
+
+    cell_2.fire_upon
+
+    assert_equal "H", cell_2.render
+    assert_equal false, cruiser.sunk?
+  end
```

Comment:
Because we are only testing against one cell, I think we can either remove one of the cells in the setup for this test OR use the second cell to assert that it has not changed.

Same for the next two test